src/content/learn/understanding-your-ui-as-a-tree.md

```diff
@@ -0,0 +1,268 @@
+<DeepDive>
+
+#### Where are the HTML tags in the render tree? {/*where-are-the-html-elements-in-the-render-tree*/}
+
+You'll notice in the above render tree, there is no mention of the HTML tags that each component renders. This is because the render tree is only composed of React [components](learn/your-first-component#components-ui-building-blocks). 
+
+React, as a UI framework, is platform agonistic. On react.dev, we showcase examples that render to the web, which uses HTML markup as its UI primitives. But a React app could just as likely render to a mobile or desktop platform, which may use different UI primitives like [UIView](https://developer.apple.com/documentation/uikit/uiview) or [Framework Element](https://learn.microsoft.com/en-us/dotnet/api/system.windows.frameworkelement?view=windowsdesktop-7.0).
+
+These platform UI primitives are not a part of React. React render trees can provide insight to our React app regardless of what platform your app renders to.
+
+</DeepDive>
+
+A render tree represents a single render pass of a React application. With [conditional rendering](/learn/conditional-rendering), a parent component may render different children depending on the data passed.
+
+We can update the Greeting app to conditionally render either an image or a text greeting.
+
+<Sandpack>
+
+```js App.js
+import { Title } from './GreetingMedia';
+import Greeting from './Greeting';
+import Copyright from './Copyright';
+
+export default function App() {
+  return (
+    <>
+      <Title text="Generate a random greeting" />
+      <Greeting>
+        <Copyright year={2004} />
+      </Greeting>
+    </>
+  );
+}
+
+```
+
+```js GreetingMedia.js
+export function Title({fancy, text}) {
+  return <h1 className={fancy ? 'fancy' : ''}>{text}</h1>;
+}
+
+export function Image({src}) {
+  return <img src={src} />;
+}
+```
+
+```js Greeting.js
+import * as React from 'react';
+import greetings from './greetings';
+import {Title, Image} from './GreetingMedia';
+
+function randIndex(max) {
+  return Math.floor(Math.random() * max);
+}
+
+export default function Greeting({children}) {
+  const [index, setIndex] = React.useState(randIndex(greetings.length));
+  const greeting = greetings[index];
+  const nextGreeting = () => setIndex(randIndex(greetings.length));
+  return (
+    <>
+      {greeting.type === 'text' ? (
+        <Title fancy text={greeting.value} />
+      ) : (
+        <Image src={greeting.value} />
+      )}
+      <button onClick={nextGreeting}>🎲</button>
+      {children}
+    </>
+  );
+}
+
+```
+
+```js Copyright.js
+export default function Copyright({year}) {
+  return <p>©️ {year}</p>;
+}
+```
+
+```js greetings.js
+export default [
+    {type: 'text', value: 'Howdy!'},
+    {type: 'text', value: 'Hello!'},
+    {type: 'image', value: 'https://i.imgur.com/yXOvdOSs.jpg'},
+    {type: 'image', value: 'https://i.imgur.com/yXOvdOSs.jpg'},
+    {type: 'text', value: 'Howdy!'},
```

Comment:
Is the second "Howdy!" on purpose?

Reply:
Nope! I need to clean this example up a bit and find different images

Reply:
Cleaned up the example and changed up the example a bit.